config-controller/internal/controller/policy_controller.go

```diff
@@ -103,44 +102,44 @@ func (r *SecurityPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 
 	// Non deletion case, so either an update or create of a policy
 	desiredState := policyCR.Spec.ToProtobuf()
-	desiredState.Name = policyCR.GetName()
-	desiredState.Source = storage.PolicySource_DECLARATIVE
+	desiredState.Id = policyCR.Status.PolicyId
 
-	existingPolicy, exists, err := r.PolicyClient.GetPolicy(ctx, req.Name)
+	_, exists, err := r.PolicyClient.GetPolicy(ctx, desiredState.GetName())
 	if err != nil {
 		return ctrl.Result{}, errors.Wrap(err, "Failed to fetch policy")
 	}
 
 	var retErr error
-	if exists {
-		desiredState.Id = existingPolicy.Id
```

Comment:
The ID needs to be pulled from `existingPolicy` when it's not yet available in the status.  Otherwise `PUT`s would fail.